### ISSUE TITLE
AGE Onboarding - Setup Developer Environment 2

### ISSUE THREAD

**Issue by JoshInnis** [apache/age#353]
1. Install Postgres from Source Code
2. Install AGE from Source Code
3. Install a Debugger
4. Learn Basic Git Commands
5. Read Regression Tests

**Comment by JoshInnis** [comment]
To debug a PostgreSQL session -

Have Postgres configured with debugging flags enable and have a Postgres instance running.
An open session running.
Know the process of the session that you will be debugging (ps -C postgres).
Execute the command gdb -p

The basic GDB commands are -
b for breakpoint, fx, b exec_simple_query or b 930
c for continue - continues to the next breakpoint
n for next line
s for step into
p for print, fx, p *node
d for delete all breakpoints
q for quit

**Comment by JoshInnis** [comment]
The Basic Git commands are:

git apply
git add
git commit
git branch
git checkout

**Comment by JoshInnis** [comment]
Install Postgres from Source Code:
https://www.postgresql.org/docs/current/install-procedure.html

Enable Debugging Flags

**Comment by JoshInnis** [comment]
Install AGE From Source Code:
https://age.apache.org/age-manual/master/intro/setup.html

**Comment by JoshInnis** [comment]
Run the Regression Tests:
https://age.apache.org/age-manual/master/contributing/run_tests.html

Then please read through regress/sql/* and regress/expected/*

To learn what the regression tests are

**Comment by Zahid07** [comment]
Hi, Nice to meet you all

**Comment by Hunaid2000** [comment]
Hi everyone. I'm excited to work with you all.

**Comment by fatimafatimaprogrammer** [comment]
Hello everyone! I'm excited and looking forward to start working on the project.

**Comment by FahadZaheerfzr** [comment]
Hey guys! Looking forward to work with you all

**Comment by Zainab-Saad** [comment]
Hey everyone, I am looking forward to work with you all

**Comment by KamleshKumar427** [comment]
Hello everyone!! Very excited to start working with you.

**Comment by moeed-k** [comment]
Hello, I'm super excited to work on this!

**Comment by imranzaheer612** [comment]
Hi everyone! Happy to be a part of this team. Let's go!

**Comment by umer200** [comment]
Hey buddies ! Lets make this journey great together !
Learn, Code, Help We move together !!

**Comment by fatimafatimaprogrammer** [comment]
> Install Postgres from Source Code: https://www.postgresql.org/docs/current/install-procedure.html
> 
> Enable Debugging Flags
I successfully installed Postgres from source code using Help from Apache AGE Discord Group

**Comment by JawadAliAbbasi** [comment]
Hi guys!

Exited and looking forward to working with you all.

**Comment by abdulmanann** [comment]
Hello everyone!

Looking to work on this project!

**Comment by TalhaMunir-JD** [comment]
Hi guys!! 

Excited to work on this

**Comment by waleedahmed0001** [comment]
Hello everyone!

Looking forward to work with you all on this project!

**Comment by JawadAliAbbasi** [comment]
Done with installation process. Would love to anyone who needs help with anything.

**Comment by titoausten** [comment]
> Run the Regression Tests: https://age.apache.org/age-manual/master/contributing/run_tests.html
> 
> Then please read through regress/sql/* and regress/expected/*
> 
> To learn what the regression tests are

Do i run the tests in the postgres directory?

**Comment by mohayu22** [comment]
@titoausten Follow these steps:

- Navigate to the directory where you have installed Apache Age.

-  Right-click in that location and select "Open Terminal."

-  Type "make installcheck" in the terminal and observe the results.

**Comment by titoausten** [comment]
> @titoausten Follow these steps:
> 
> - Navigate to the directory where you have installed Apache Age.
> 
> -  Right-click in that location and select "Open Terminal."
> 
> -  Type "make installcheck" in the terminal and observe the results.
> 

Thank you, I'll do that and get back to you on the result.

**Comment by titoausten** [comment]
![Image](https://user-images.githubusercontent.com/94079649/222970484-89594b8b-50ae-40ec-8e7e-0b3d90124e84.png)

@mohayu22 this is the result. How do I go about it?

**Comment by Hunaid2000** [comment]
> ![Image](https://user-images.githubusercontent.com/94079649/222970484-89594b8b-50ae-40ec-8e7e-0b3d90124e84.png)
> 
> @mohayu22 this is the result. How do I go about it?

Have you installed Postgres? If yes, you need to provide path to pg_config. Check this [video](https://www.youtube.com/watch?v=0-qMwpDh0CA) for complete process.

**Comment by titoausten** [comment]
> @titoausten Have you installed Postgres? If yes, you need to provide path to pg_config like `make PG_CONFIG=<pg-install-dir>/bin/pgconfig installcheck`. Check this [video](https://www.youtube.com/watch?v=0-qMwpDh0CA) for complete process.

Yes, thank you. I noticed it was the Variables always missing from the PATH anytime i ended a session. Well I guess I have to go about it using relative path.

**Comment by dukeofhazardz** [comment]
> @titoausten Have you installed Postgres? If yes, you need to provide path to pg_config like `make PG_CONFIG=<pg-install-dir>/bin/pgconfig installcheck`. Check this [video](https://www.youtube.com/watch?v=0-qMwpDh0CA) for complete process.
This was super helpful, thank you so much

**Comment by mohanadtoaima** [comment]
Hi guys, should I procure an EC2 RHEL8 instance to install postgreSQL or what should I do ?

**Comment by dukeofhazardz** [comment]
> install postgreSQL

You can, if you want to host your database server on there
But no, it is not necessary to use an EC2 RHEL8 instance to install PostgreSQL.

**Comment by dukeofhazardz** [comment]
I think that it is also important to ensure you have flex and bison installed on your machine before running PG_CONFIG to save yourself the stress of Postgre reconfiguration or even reinstallation

**Comment by DavidGMdev** [comment]
Hello everyone, 
Happy to work with you all !

**Comment by MuhammadZeeshan03** [comment]
Hello Everyone

Excited to work with you all on this project.

**Comment by HRmemon** [comment]
Hi guys!

Excited to work with you all!

**Comment by Muzzammil242** [comment]
Hello guys!

Excited to work with you all!

**Comment by Hannan2910** [comment]
Hello guys!

Really excited to work with you all

**Comment by abdullahzia-6464** [comment]
Excited to work on the project with everyone!

**Comment by salarabbasi01** [comment]
Hello Everyone
Excited to be working with you all

**Comment by Jahangirbabar2000** [comment]
Hi everyone! excited to work with you all during this internship!

**Comment by Ammar-Baig19** [comment]
Great, learning lots of new things. Excited to work with you all during this internship!

**Comment by fahadullah-112** [comment]
Very informative and Great, learning lots of new things. Excited to work with you all during this internship.
